x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/SemanticTextInferenceProcessor.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.inference.ingest;
+
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.ingest.AbstractProcessor;
+import org.elasticsearch.ingest.CompoundProcessor;
+import org.elasticsearch.ingest.IngestDocument;
+import org.elasticsearch.ingest.Processor;
+import org.elasticsearch.ingest.WrappingProcessor;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.TextExpansionConfigUpdate;
+import org.elasticsearch.xpack.ml.notifications.InferenceAuditor;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+public class SemanticTextInferenceProcessor extends AbstractProcessor implements WrappingProcessor {
```

Comment:
A new inference processor that uses internally the InferenceProcessor. It simplifies creating an ingestion pipeline.